File: src/main/java/mk/ukim/finki/wp/emt_lab/model/domain/Book.java
package mk.ukim.finki.wp.emt_lab.model.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import mk.ukim.finki.wp.emt_lab.model.enums.Category;
import mk.ukim.finki.wp.emt_lab.model.enums.State;

import java.time.LocalDate;
import java.time.LocalDateTime;


@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "books")
public class Book extends BaseAuditableEntity {

    @Column(nullable = false)
    private Boolean deleted = false;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private Category category;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private State state;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    private Author author;
    private LocalDateTime date_published;


    public Book(String name, Category category, State state,LocalDateTime date_published, Author author) {
        this.name = name;
        this.category = category;
        this.state = state;
        this.author = author;
        this.deleted = false;
        this.date_published=date_published;
    }
}

File: frontend/src/api/types/book.ts
export interface Book {
    id: number;
    name: string;
    category: string;
    state: string;
    authorId: number;
    authorName: string;
    availableCopies: number;
}

export interface CreateBookDto {
    name: string;
    category: string;
    state: string;
    authorId: number;
    availableCopies: number;
}

export interface UpdateBookDto {
    name: string;
    category: string;
    state: string;
    authorId: number;
    availableCopies: number;
}